packages/esm-service-queues-app/src/createDashboardLink.component.tsx

```diff
@@ -22,13 +25,25 @@ function DashboardExtension({ dashboardLinkConfig }: { dashboardLinkConfig: Dash
   }, [location.pathname]);
 
   return (
-    <ConfigurableLink
-      className={classNames('cds--side-nav__link', {
-        'active-left-nav-link': navLink.match(name),
-      })}
-      to={`${spaBasePath}/${name}`}>
-      {t('serviceQueues', 'Service queues')}
-    </ConfigurableLink>
+    <div>
+      <ConfigurableLink
+        className={classNames('cds--side-nav__link', {
+          'active-left-nav-link': navLink.match(name),
+        })}
+        to={`${spaBasePath}/${name}`}>
+        {config.showDashboardLinkIcon ? (
+          <Button
+            className={styles.navlinkIcon}
+            hasIconOnly
+            size="sm"
+            kind="ghost"
+            renderIcon={renderIcon}
+            iconDescription="Add"
+          />
+        ) : null}{' '}
```

Comment:
This might be a stretch but i was of the view that we build this into `ConfigurableLink` component and we can pass the Icon name or component instead of adding a button component every time we need to add an icon. 

Carbon has provision for `renderIcon` within the `SideNavMenu` component ([Docs Link](https://react.carbondesignsystem.com/?path=/story/components-ui-shell-sidenav--fixed-side-nav-w-icons)) but this doesn't seem like the path we took to add the links so we could have provision for something similar like so 

```
renderIcon={Fade}
```

@ibacher @denniskigen what's your view on this?

Reply:
I think that this would somewhat confuse the purpose of `ConfigurableLink`. It is not really a component for rendering links in the side navigation menu. It is a component for rendering links in any context. I'd possibly be on-board with a component that combined a `ConfigurableLink` with a button, but that would basically be the `DashboardExtension` that I suggested above might make sense to be its own component.

Reply:
I have created this component to sort this out : https://github.com/openmrs/openmrs-esm-core/pull/885